### ISSUE TITLE
unittest ValueError when invoking as module

### ISSUE THREAD

**Issue by ed1c8aea-5883-48b8-a036-3168f6e39e01** [python/cpython#90579]
BPO | [46421](https://bugs.python.org/issue46421)
--- | :---
Nosy | @rbtcollins, @ezio-melotti, @voidspace, @serhiy-storchaka, @JelleZijlstra, @miss-islington, @kumaraditya303, @BaderSZ
PRs | <li>python/cpython#30654</li><li>python/cpython#31969</li><li>python/cpython#31970</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2022-03-18.03:25:14.619>
created_at = <Date 2022-01-17.23:35:01.985>
labels = ['type-bug', '3.8', '3.9', '3.10', '3.11', '3.7', 'library']
title = 'unittest ValueError when invoking as module'
updated_at = <Date 2022-03-18.03:25:14.618>
user = 'https://github.com/BaderSZ'
```

bugs.python.org fields:
```python
activity = <Date 2022-03-18.03:25:14.618>
actor = 'JelleZijlstra'
assignee = 'none'
closed = True
closed_date = <Date 2022-03-18.03:25:14.619>
closer = 'JelleZijlstra'
components = ['Library (Lib)']
creation = <Date 2022-01-17.23:35:01.985>
creator = 'BaderSZ'
dependencies = []
files = []
hgrepos = []
issue_num = 46421
keywords = ['patch']
message_count = 5.0
messages = ['410833', '410834', '415461', '415462', '415468']
nosy_count = 8.0
nosy_names = ['rbcollins', 'ezio.melotti', 'michael.foord', 'serhiy.storchaka', 'JelleZijlstra', 'miss-islington', 'kumaraditya', 'BaderSZ']
pr_nums = ['30654', '31969', '31970']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue46421'
versions = ['Python 3.7', 'Python 3.8', 'Python 3.9', 'Python 3.10', 'Python 3.11']
```

</p></details>

**Comment by AliRn76** [comment]
I'm not sure what is wrong but I still have this issue on windows, tell me if you need any more information
(The log below is from github workflow `windows-latest`)

```
Run python -m unittest .\tests\test_*.py
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\unittest\__main__.py", line 18, in <module>
    main(module=None)
  File "C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\unittest\main.py", line 100, in __init__
    self.parseArgs(argv)
  File "C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\unittest\main.py", line 147, in parseArgs
    self.createTests()
  File "C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\unittest\main.py", line 158, in createTests
    self.test = self.testLoader.loadTestsFromNames(self.testNames,
  File "C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\unittest\loader.py", line 220, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\unittest\loader.py", line 220, in <listcomp>
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\unittest\loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
ValueError: Empty module name
```

**Comment by James-E-A** [comment]
I see this was supposed to have been fixed in a0db11b10fca0fee6bb2b8d6277e266bad8c0fdb, but I believe I'm still experiencing this issue on Python 3.11.9.

https://github.com/James-E-A/temp_20313
